client/app/visualizations/word-cloud/Editor.jsx

```diff
@@ -1,30 +1,95 @@
-import { map } from 'lodash';
+import { map, merge } from 'lodash';
 import React from 'react';
 import Select from 'antd/lib/select';
+import InputNumber from 'antd/lib/input-number';
+import * as Grid from 'antd/lib/grid';
 import { EditorPropTypes } from '@/visualizations';
 
-const { Option } = Select;
-
 export default function Editor({ options, data, onOptionsChange }) {
-  const onColumnChanged = (column) => {
-    const newOptions = { ...options, column };
-    onOptionsChange(newOptions);
+  const optionsChanged = (newOptions) => {
+    onOptionsChange(merge({}, options, newOptions));
   };
 
   return (
-    <div className="form-group">
-      <label className="control-label" htmlFor="word-cloud-column">Word Cloud Column Name</label>
-      <Select
-        id="word-cloud-column"
-        className="w-100"
-        value={options.column}
-        onChange={onColumnChanged}
-      >
-        {map(data.columns, ({ name }) => (
-          <Option key={name}>{name}</Option>
-        ))}
-      </Select>
-    </div>
+    <React.Fragment>
+      <div className="form-group">
+        <label className="control-label" htmlFor="word-cloud-words-column">Words Column</label>
+        <Select
+          id="word-cloud-words-column"
+          className="w-100"
+          value={options.column}
+          onChange={column => optionsChanged({ column })}
+        >
+          {map(data.columns, ({ name }) => (
+            <Select.Option key={name}>{name}</Select.Option>
+          ))}
+        </Select>
+      </div>
+      <div className="form-group">
+        <label className="control-label" htmlFor="word-cloud-frequencies-column">Frequencies Column</label>
+        <Select
+          id="word-cloud-frequencies-column"
+          className="w-100"
+          value={options.frequenciesColumn}
+          onChange={frequenciesColumn => optionsChanged({ frequenciesColumn })}
+        >
+          <Select.Option key="none" value=""><i>(count word frequencies automatically)</i></Select.Option>
+          {map(data.columns, ({ name }) => (
+            <Select.Option key={'column-' + name} value={name}>{name}</Select.Option>
+          ))}
+        </Select>
+      </div>
+      <div className="form-group">
+        <label className="control-label" htmlFor="word-cloud-word-length-limit">
+          Words Length Limit
```

Comment:
So if I understand correctly, it's a limit range and the fields are "from" and "to"?

Reply:
Yes, correct

Reply:
In that case, the UI is a bit unclear. How about a slider instead? https://ant.design/components/slider/

Reply:
(Or descriptive labels next to the numberinputs)

Reply:
Not sure if slider is a good choice because technically there is no upper limit (user can use any positive number which even may be larger than any value from dataset); also, lower limit is `1` and there is "null" value (no limit). Maybe, instead of slider and labels I can set placeholders to `min`/`max` (instead of current `no limit`)?

Reply:
@ranbena Done (this and previous suggestion)